File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    visited = [0 for _ in range(n)]
    answer = 0

    def bfs(idx):
        Q = deque()
        Q.append(idx)
        visited[idx] = 1
        
        while Q:
            c = Q.popleft()
            for next in range(n):
                if computers[c][next] == 1 and visited[next] == 0:
                    visited[next] = 1
                    Q.append(next)
                    
    for i in range(n):
        if visited[i] == 0:
            answer += 1
            bfs(i)
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    const visited = Array(n).fill(0);
    let answer = 0;
    
    function bfs(start) {
        const queue = [];
        queue.push(start);
        visited[start] = 1
        
        while(queue.length > 0) {
            c = queue.shift();
            
            for (j = 0; j < n; j ++) {
                if (!visited[j] && computers[c][j] === 1) {
                    visited[j] = 1
                    queue.push(j)
                }
            }
        }
    }
    
    for (i = 0; i < n; i ++) {
        if (!visited[i]) {
            answer += 1;
            bfs(i)
        }
    }
    
    return answer;
}